### PR TITLE
feat: Add basic user checklist main page

### LINKED ISSUE #140: feat: Basic user checklist main page
## Description

User main checklist page, the "list of checklists" if you will. 

Includes the following requirements:

- 2 tabs: To Be Done and Completed
- view when all checklists are completed
- start button for lists not yet started and continue button for lists in progress

## UI

Screenshot 2024-10-22 at 4 32 14 PM

Screenshot 2024-10-22 at 4 32 51 PM
Continue button for lists in progress

Screenshot 2024-10-22 at 4 33 27 PM

Screenshot 2024-10-22 at 4 33 37 PM

### PR DESCRIPTION
**Pull request by hinxcode**
This pull request introduces a comprehensive checklist feature to the Support Sphere app, including new models, repository, service, cubit, and UI components. The changes are organized into several themes: adding new models, updating the repository and service, implementing state management with Cubit, and updating the UI.

### New Models:
* [`src/support_sphere/lib/data/models/checklist.dart`](diffhunk://#diff-7d28a60a377ce6079cee0c1b7095523994c05db01a769d57104fd5bf8d15a569R1-R61): Added `Checklist` and `ChecklistSteps` models to represent checklists and their steps.
* [`src/support_sphere/lib/data/models/frequency.dart`](diffhunk://#diff-a0c5862e1a466fa27c60ae510e0d1e99b60905863bb80170864bc5487c546d66R1-R16): Added `Frequency` model to represent the frequency of checklists.

### Repository and Service:
* [`src/support_sphere/lib/data/repositories/checklist.dart`](diffhunk://#diff-19f84eecb3a5980cec787bf65f0a186bedbbc9f8492937f509f3bc52ed7a0a7cR1-R46): Created `ChecklistRepository` to handle checklist-related data operations.
* [`src/support_sphere/lib/data/services/checklist_service.dart`](diffhunk://#diff-089211716db7a731e641763326f555ae56a77ce0059d0f0c4b43a51f22244007R1-R48): Created `ChecklistService` to interact with the Supabase client and fetch user checklists.

### State Management:
* [`src/support_sphere/lib/logic/cubit/checklist_cubit.dart`](diffhunk://#diff-043e8341304beb1c94760d3292d0d5a53544b10b6ce7b744d20c96e975eb163dR1-R33): Implemented `ChecklistCubit` to manage the state of checklists.
* [`src/support_sphere/lib/logic/cubit/checklist_state.dart`](diffhunk://#diff-dd572e6389e1872ebed95c998e718b46d3585a2c3e68e09d6bcb6b955a156aecR1-R25): Defined `ChecklistState` to handle the state of the checklist main page.

### UI Components:
* [`src/support_sphere/lib/presentation/components/checklist_card.dart`](diffhunk://#diff-0dd94d14519299af52ab10acdf001dc6fba175c46bf5e04c0281a07a285a95e3R1-R96): Added `ChecklistCard` widget to display individual checklists.
* [`src/support_sphere/lib/presentation/pages/main_app/checklist/checklist_main_body.dart`](diffhunk://#diff-ea40a36ab01761c090096310715183df61698c41acff39fce8972f3d9cc02af9R1-R169): Implemented `ChecklistBody` and its associated tabs for displaying to-be-done and completed checklists.

### Other Changes:
* [`src/support_sphere/lib/constants/routes.dart`](diffhunk://#diff-cbff05c7ffb5e17adbd5bd2b934d0b73a29d29f625d413778ae7dfa43cda9d39R6): Updated routes to include the new checklist feature.
* [`src/support_sphere/lib/constants/string_catalog.dart`](diffhunk://#diff-9663c9e5e9c15a602ae60e64b7178cbfd727abbac179b5aaefcab70a956373fdL4-R20): Added new strings for checklist messages and reformatted existing strings for better readability.
* [`src/support_sphere/pubspec.yaml`](diffhunk://#diff-4db6703e1a00162c625ec19e59f4ac28dee1c4bff679671a04cd257810e19262R56): Added `intl` dependency for date formatting.

### Related Issue
Resolves #140 

Regarding the original UI requirements, I made some adjustments. Instead of start & continue buttons, I enabled the tab feature on the cards themselves, which is more intuitive and aligned with the material design.

I inserted sample data into `frequency` table manually, which can be written in scripts in the future:
Screenshot 2024-11-18 at 00 47 26

### DEMO
Screenshot 2024-11-18 at 00 54 56
Screenshot 2024-11-18 at 00 48 31
Screenshot 2024-11-18 at 00 48 07
Screenshot 2024-11-18 at 00 48 14
Screenshot 2024-11-18 at 00 45 43
Screenshot 2024-11-18 at 00 45 51

### CODE REVIEW

**Comment by nikiburggraf** [comment]
This view should be in the "Prepare" tab, not "Manage Checklists"

**Comment by lsetiawan** [comment]
@hinxcode and @nikiburggraf checking in, what is the status of this PR? Is this ready for another review?

**Comment by hinxcode** [comment]
> @hinxcode and @nikiburggraf checking in, what is the status of this PR? Is this ready for another review?

Hi Don! I’ve updated the PR, and it’s now ready for your review.

File: src/support_sphere/lib/constants/string_catalog.dart
Comment (hinxcode) [inline src/support_sphere/lib/constants/string_catalog.dart:1]: I used IDE formatting, so some other lines have also been modified.

**Review by lsetiawan — APPROVED** [review]
Thanks @hinxcode. I tried this locally and looked through the code and it looks great!

**Pull request merged**